### PR TITLE
feat(integration-directory): update project select modal

### PR DESCRIPTION
**Pull request by scefali**
This PR changes the project/org select modal to use the design from Chris without the changes to the modal header. As a team, we decided that we should tackle the modal header change globally and not change this particular one in isolation. The main stylistic change is the border of the items in the dropdown.

I made the text dynamic based on whether the user has to pick a project and/or an organization. I've also removed the `deprecatedSelectControl` prop so I can apply custom styling to the dropdown.

Before:
Screen Shot 2020-02-26 at 2 39 26 PM

After:
Screen Shot 2020-02-26 at 3 52 38 PM

Note I need to update tests which are broken

### CODE REVIEW

**Comment by billyvg** [comment]
This looks much better imo, but now this modal is inconsistent with our other modals, cc @getsentry/design

**Comment by scefali** [comment]
@billyvg Yea, I had that concern as well. Hence the discussion in discuss-frontend this morning. Not sure what the best way forward is with this.

**Comment by scefali** [comment]
@billyvg reverted the change for the modal header so now it matches the rest of our modals

**Review by billyvg — APPROVED** [review]
Damn, the other style was much cleaner

**Comment by scefali** [comment]
@billyvg Yea, we should implement the modal globally. Probably worth talking about in the frontend TSC tomorrow, although I won't be able to make it.

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 1]
```diff
@@ -228,15 +244,24 @@ class ContextPickerModal extends React.Component<Props> {
...
       .map(({slug}) => ({label: slug, value: slug}));
 
+    const customStyles = {
+      option: provided => ({
+        ...provided,
+        borderTop: `1px solid ${theme.gray1}`,
+      }),
+      menuList: provided => ({
+        ...provided,
+        paddingTop: '0px',
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx]:
```suggestion
        paddingTop: '0',
```
0 doesn't need units :golf:

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx (hunk 1 above)
Comment (scefali) [reply src/sentry/static/sentry/app/components/contextPickerModal.tsx]: @markstory  ooops...merged before making this change. I'll remember this next time.

**Comment by benvinegar** [comment]
@scefali How do we get platform icons into this modal?

**Comment by scefali** [comment]
@benvinegar Good callout. I had to roll this PR back since it caused a minor bug (2 impacted users) so I can add it in my redo.

**Pull request merged**